amazon_cloud_code_generator/data/plugins/module_utils/core.py

```diff
@@ -469,10 +476,10 @@ def update_resource(
                 ) as e:
                     self.module.fail_json_aws(e, msg="Failed to update resource")
 
-                if self.module.params.get("wait"):
-                    self.wait_until_resource_request_success(
-                        response["ProgressEvent"]["RequestToken"]
-                    )
+                # Always wait to update to ensure correct data returned to user
+                self.wait_until_resource_request_success(
+                    response["ProgressEvent"]["RequestToken"]
```

Comment:
We should always wait for the resource to be updated.

Reply:
@alinabuzachis if we dont wait, then the data returned to the user can often be incorrect still the update is still in progress. We can choose to not wait, but if we want to return accurate data, we would need to track a `correct` resource to return, rather than using the `get_resource` function without waiting